src/fileNamer.js

```diff
@@ -151,3 +151,139 @@ function mapValue(map, value) {
+        log('info', `New label: ${doc.label}`)
+      }
+      log('warn', 'Form Unknown')
+      failed++
+      continue
+      //throw 'Form Unknown'
+    }
+
+    doc.oldname =
+      `${doc.year}-${doc.category}-${doc.type}` +
+      (doc.form === '' ? '-' : `-${doc.form}-`)
+    if (LWP) {
+      log('info', '-----')
+      log('info', `New label: ${doc.label}`)
+      log('info', `Old name prediction: ${doc.oldname}`)
+      log('info', '-----')
+    }
+  }
+  log(
+    'info',
+    `Failed category, type or form evaluation: ${failed} on ${documents.length}`
+  )
+  log('info', `#Failed#: ${((100 * failed) / documents.length).toFixed(2)}%`)
+  return documents
+}
+
+function evalCategory(doc) {
+  if (doc.label.match(/revenus/)) {
+    return 'ImpotsRevenus'
+  } else if (doc.label.match(/habitation/)) {
+    return 'TaxeHabitation'
+  } else if (doc.label.match(/foncières/)) {
+    return 'TaxeFoncière'
+  } else if (doc.label.match(/RedevanceAudiovisuelle/)) {
+    // WARNING, NEVER ENCOUNTERED
+    return 'RedevanceAudiovisuelle'
+  } else {
+    return undefined
+  }
+}
+
+function evalType(doc) {
+  if (
+    doc.label.match(/^Avis d'impôt/) ||
+    doc.label.match(/^Avis de/) ||
+    doc.label.match(/^1er Avis/) ||
+    doc.label.match(/^2e Avis/) ||
+    doc.label.match(/^Avis échéancier/)
+  ) {
+    return 'Avis'
+  } else if (
+    doc.label.match(/^Déclaration en ligne/) ||
+    doc.label.match(/^Déclaration complémentaire/) ||
+    doc.label.match(/^Déclaration/) // recoupe les 2 autres
+  ) {
+    return 'Formulaire'
+  } else if (doc.label.match(/^Accusé de réception/)) {
+    return 'AccuséRéception'
+  } else {
+    return undefined
+  }
+}
+
+function evalForm(doc, year, category) {
+  if (category === 'ImpotsRevenus') {
+    if (doc.label.match(/^Accusé de réception/)) {
+      return ''
+    } else if (doc.label.match(/^Avis d'impôt/)) {
+      return '1'
+    } else if (doc.label.match(/^1er Avis d'Acompte/)) {
+      return '811'
+    } else if (doc.label.match(/^2e Avis d'Acompte/)) {
+      return '812'
+    } else if (doc.label.match(/^Avis de situation déclarative/)) {
+      return 'SituationDéclarative'
+    } else if (doc.label.match(/^Déclaration en ligne \d{4} de revenus \(/)) {
+      return '2042'
+    } else if (
+      doc.label.match(
+        //eslint-disable-next-line
+          /^Déclaration en ligne 2019 de revenus \: réductions et crédits d'impôt \(/
+      )
+    ) {
+      return '2042RICI'
+    }
+  }
+
+  // Taxe d'habitation
+  if (category === 'TaxeHabitation') {
+    if (doc.label.match(/^Avis de taxe d'habitation/)) {
+      if (year >= 2019) {
+        return ''
+      } else {
+        return 'Primitif'
+      }
+    } else if (doc.label.match(/^Avis échéancier/)) {
+      return 'Echeancier'
+    }
+  }
+
+  // Taxe foncière
+  if (category === 'TaxeFoncière') {
+    if (doc.label.match(/^Avis de taxes foncières/)) {
+      return 'Primitif'
+    }
+  }
+  // Catch unknown form
```

Comment:
This comment is not very informative, I would remove it.

Reply:
Phrasing changed to be more accurate